core/domain/collection_domain.py

```diff
@@ -1145,6 +1152,35 @@ def delete_node(self, exploration_id):
                 exploration_id)
         del self.nodes[node_index]
 
+    def move_node(self, exploration_id, new_position):
+        """Changes the position of the node corresponding to the given
+        exploration in the nodes list of the collection.
+
+        Args:
+            exploration_id: str. The id of the exploration.
+            new_postion: int. The new position of the node in the nodes list.
+
+        Raises:
+            ValueError: The exploration is not part of the collection.
+            ValueError: The exploration cannot be moved to the new_position or
```

Comment:
Could you be more specific as to which situations would lead to the exploration not being able to be moved? 'cannot be moved' is a bit vague, so if the ValueError occurs it may not be obvious (depending on the error's message) what went wrong or how to fix it.

Reply:
Also as a follow-up: Given that 'ValueError' is the only type of exception raised, couldn't we just list it 1 time and then enumerate all the different cases where it's thrown? I.e. due to it not being part of the collection, already being at the position, etc.

Reply:
FYI, re the second follow-up, we do this elsewhere in the codebase, so I think listing ValueError multiple times is fine.

Will defer other questions to @DubeySandeep. Thanks!